dashboard-api/internal/platform/storage/cockroachdb/study_repository_test.go

```diff
@@ -76,3 +76,74 @@ func Test_StudyRepository_GetStudyBySlug_Succeed(t *testing.T) {
 	expectedStudy := studiesmanager.NewStudy("7261456a-77b7-4731-a499-629f9f49abe8", "Example Study", "example-study", currentTime)
 	assert.Equal(t, expectedStudy, study)
 }
+
+func Test_StudyRepository_GetStudies_UnexpectedError_during_Query(t *testing.T) {
+	offset := 0
+	limit := 20
+
+	db, sqlMock, err := sqlmock.New(sqlmock.QueryMatcherOption(sqlmock.QueryMatcherEqual))
+	require.NoError(t, err)
+
+	sqlMock.ExpectQuery(
+		"SELECT id, name, slug, created FROM studies ORDER BY created DESC OFFSET $1 LIMIT $2").
```

Comment:
Don't repeat business logic of the unit under test in the test itself! A template and the correct args to fill it make up business logic in this case.

Reply:
I'm aware of this but the mocking library forces you to write it like this.

The following is not possible:
```
sqlMock.ExpectQuery(
		"SELECT id, name, slug, created FROM studies ORDER BY created DESC OFFSET 0 LIMIT 20").
		WillReturnError(errors.New("unexpected-error"))
```